homeassistant/components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,390 @@
+        if not self.values.mode:
+            return []
+        # Z-Wave uses mode-values > 10 for presets
+        return [PRESET_NONE] + [
+            val[VALUE_LABEL]
+            for val in self.values.mode.value[VALUE_LIST]
+            if val[VALUE_ID] > 10
+        ]
+
+    @property
+    def swing_mode(self):
+        """Return the swing setting."""
+        if not self.values.swing_mode:
+            return None
+        return self.values.swing_mode.value
+
+    @property
+    def swing_modes(self):
+        """Return the list of available swing modes."""
+        if not self.values.swing_mode:
+            return []
+        return [
+            entry[VALUE_LABEL] for entry in self.values.swing_mode.value[VALUE_LIST]
+        ]
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) != 1:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the lowbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the highbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[1].value
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature.
+
+        Must know if single or double setpoint.
+        """
+        if len(self._current_mode_setpoint_values) == 1:
+            setpoint = self._current_mode_setpoint_values[0]
+            target_temp = kwargs.get(ATTR_TEMPERATURE)
+            if setpoint is not None and target_temp is not None:
+                setpoint.send_value(target_temp)
+        elif len(self._current_mode_setpoint_values) == 2:
+            (setpoint_low, setpoint_high) = self._current_mode_setpoint_values
+            target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+            if setpoint_low is not None and target_temp_low is not None:
+                setpoint_low.send_value(target_temp_low)
+            if setpoint_high is not None and target_temp_high is not None:
+                setpoint_high.send_value(target_temp_high)
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        if not self.values.fan_mode:
+            return
+        # get id for this fan_mode
+        fan_mode_value = None
+        for val in self.values.fan_mode.value[VALUE_LIST]:
+            if val[VALUE_LABEL] == fan_mode:
+                fan_mode_value = val[VALUE_ID]
```

Comment:
Should we break if we find our target?
```suggestion
                fan_mode_value = val[VALUE_ID]
                break
```